R/waitUntilFinished.R

```diff
@@ -15,11 +17,33 @@
 #' check(job) # should be complete
 #' }
 #' @export
-wait <- function(.Object, sleep.time = gconfig('sleep.time')){
+setGeneric(name="wait",def=function(.Object, sleep.time){standardGeneric("wait")})
+
+#' @rdname wait
+#' @aliases wait
+setMethod(f = "wait",signature(.Object = "geojob", sleep.time = "numeric"), definition = function(.Object, sleep.time){
+  wait(id(.Object), sleep.time = sleep.time)
+})
+
+#' @rdname wait
+#' @aliases wait
+setMethod(f = "wait",signature(.Object = "geojob", sleep.time = "missing"), definition = function(.Object, sleep.time){
+  wait(id(.Object), sleep.time = gconfig('sleep.time'))
+})
+
+#' @rdname wait
+#' @aliases wait
+setMethod(f = "wait",signature(.Object = "character", sleep.time = "numeric"), definition = function(.Object, sleep.time){
   running <- running(.Object, retry = TRUE)
   while(running){
     Sys.sleep(sleep.time)
     running <- running(.Object, retry = TRUE)
   }
   invisible(.Object)
-}
+})
+
+#' @rdname wait
+#' @aliases wait
+setMethod(f = "wait",signature(.Object = "character", sleep.time = "missing"), definition = function(.Object, sleep.time){
```

Comment:
We only need to define types for signatures that we need to dispatch based on. Alternatively, I could use "ANY" for the sleep.time signature (or leave that off) and handle `if(missing(sleep.time))` within the function